### PR TITLE
fix model validation failure when path parameter value containing sep…

### PR DESCRIPTION
**Pull request by jianyexi**
…ecial charaters

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "oav",
-  "version": "0.21.3",
+  "version": "0.21.4",
```
Comment (raych1) [inline package.json:3]: Can you update 'ms-rest' version in this file?

File: lib/validators/modelValidator.ts [hunk 2]
```diff
@@ -790,10 +792,9 @@ export class ModelValidator extends SpecValidator<SpecValidationResult> {
           // validation of path segment.
           parameterValue = parameterValue.replace(/\//gi, "")
 
-          // replacing scheme that may cause validator failed when x-ms-parameterized-host enbaled
-          if (hostTemplate) {
-            // TODO: we can get the scheme from parameterValue if the useSchemePrefix is setting false in the x-ms-parameterized-host,
-            // then check if it can match to the swagger scheme.
+          // replacing scheme that may cause validator failed when x-ms-parameterized-host enbaled & useSchemePrefix enabled
+          // because if useSchemePrefix enabled ,the parameter value in x-ms-parameterized-host should not has the scheme (http://|https://)
+          if (useSchemePrefix) {
```
Comment (raych1) [inline lib/validators/modelValidator.ts:797]: useSchemePrefix is true by default if it's not set. 
https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-parameterized-host

Can you also add test case when useSchemePrefix = false?

Comment (jianyexi) [reply lib/validators/modelValidator.ts:797]: sure, that's important

Comment (jianyexi) [reply lib/validators/modelValidator.ts:797]: already has the test case in  describe : "for parameters in x-ms-parameterized-host"

File: test/modelValidatorTests.ts [hunk 3]
```diff
@@ -264,6 +264,22 @@ describe("Model Validation", () => {
       assert(result.length === 0, `swagger "${specPath2}" contains model validation errors.`)
       // console.log(result)
     })
+
+    it("should pass for parameters include pattern", async () => {
```
Comment (raych1) [inline test/modelValidatorTests.ts:268]: can you add one more test for useSchemePrefix:false?

File: regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap [hunk 4]
```diff
@@ -13287,224 +13135,12 @@ exports[`validateExamples should not regress for file '/home/vsts/work/1/s/regre
...
-    },
-    "operationId": "Alerts_UpdateResourceGroupLevelAlertState",
-    "responseCode": "ALL",
-    "scenario": "Update security alert state on a resource group from a security data location",
-    "severity": 0,
-    "source": "request",
-  },
-]
-`;
+exports[`validateExamples should not regress for file '/home/vsts/work/1/s/regression/azure-rest-api-specs/specification/security/resource-manager/Microsoft.Security/stable/2019-01-01/alerts.json': returned results 1`] = `Array []`;
```
Comment (raych1) [inline regression/__snapshots__/validateExamplesRegression_latestOnly.test.ts.snap:13138]: Do you know why this new one is added?

**Pull request merged**